pkg/accounting/accounting.go

```diff
@@ -216,6 +225,37 @@ func (a *Accounting) getPeerBalance(peer swarm.Address) (*PeerBalance, error) {
 	return peerBalance, nil
 }
 
+func (a *Accounting) getPeersBalances() (map[string]int64, error) {
+	peersBalances := make(map[string]int64)
+
+	a.balancesMu.Lock()
+	for peer, balance := range a.balances {
+		peersBalances[peer] = balance.balance
+	}
+	a.balancesMu.Unlock()
+
+	err := a.store.Iterate(balancesPrefix, func(key, val []byte) (stop bool, err error) {
+		k := string(key)
+		split := strings.SplitAfter(k, balancesPrefix)
+		if len(split) != 2 {
+			return false, fmt.Errorf("invalid overlay key: %s", k)
+		}
+		addr, err := swarm.ParseHexAddress(split[1])
+
+		if err != nil {
+			return false, err
+		}
+
+		if val, ok := peersBalances[addr.String()]; !ok {
```

Comment:
this is kind of sketchy. it would work in most cases but it questions the status of the persisted balances and whether they have precedence at all. i find it not so great that there could be some value mismatch between the in-memory and persisted value, and it will be silently ignored here. this is definitely something we need to review, but i guess we can live with this as-is for now.

Reply:
our idea was (with debouncing in mind) that in-memory values are the source of truth, and we just append stuff from store that we don't have in memory